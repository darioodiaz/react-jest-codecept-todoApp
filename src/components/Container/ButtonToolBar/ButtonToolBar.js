import React from 'react';

function ButtonToolBar(props) {
    let buttons = [
        <input type="button" name="btn-add" key="0" onClick={props.onAdd} value="Add" />, 
        <input type="button" name="btn-clear" key="1" value="Clear" onClick={props.onClear} />];
    if (props.showRemove) {
        buttons.push(<input type="button" name="btn-remove" key="2" value="Remove" onClick={props.onRemove} />);
    }
    return (<div className="button-toolbar">{buttons}</div>);
}

export default ButtonToolBar;