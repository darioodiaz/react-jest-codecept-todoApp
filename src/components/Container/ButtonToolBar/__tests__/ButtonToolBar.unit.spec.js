import React from 'react';
import ReactDOM from 'react-dom';
import ButtonToolBar from '../ButtonToolBar';

it('montar el componente sin problemas', () => {
    shallow(<ButtonToolBar onAdd={jest.fn()} onClear={jest.fn()} onRemove={jest.fn()} />);
});

it('no mostrar el boton Clear', () => {
    const el = shallow(<ButtonToolBar onAdd={jest.fn()} onClear={jest.fn()} onRemove={jest.fn()} />);
    const buttons = el.find('input[type="button"]').length;
    expect(buttons).toBe(2);
});

it('no mostrar el boton Clear con la propiedad showRemove en false', () => {
    const el = shallow(<ButtonToolBar showRemove={false} onAdd={jest.fn()} onClear={jest.fn()} onRemove={jest.fn()} />);
    const buttons = el.find('input[type="button"]').length;
    expect(buttons).toBe(2);
});

it('mostrar el boton Clear con la propiedad showRemove en true', () => {
    const el = shallow(<ButtonToolBar showRemove={true} onAdd={jest.fn()} onClear={jest.fn()} onRemove={jest.fn()} />);
    const buttons = el.find('input[type="button"]').length;
    expect(buttons).toBe(3);
});