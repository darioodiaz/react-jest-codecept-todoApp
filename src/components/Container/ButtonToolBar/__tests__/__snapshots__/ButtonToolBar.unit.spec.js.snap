exports[`test mostrar el boton Clear con la propiedad showRemove en true 1`] = `
<div
  className="button-toolbar">
  <input
    name="btn-add"
    onClick={[Function]}
    type="button"
    value="Add" />
  <input
    name="btn-clear"
    onClick={[Function]}
    type="button"
    value="Clear" />
  <input
    name="btn-remove"
    onClick={[Function]}
    type="button"
    value="Remove" />
</div>
`;

exports[`test no mostrar el boton Clear 1`] = `
<div
  className="button-toolbar">
  <input
    name="btn-add"
    onClick={[Function]}
    type="button"
    value="Add" />
  <input
    name="btn-clear"
    onClick={[Function]}
    type="button"
    value="Clear" />
</div>
`;

exports[`test no mostrar el boton Clear con la propiedad showRemove en false 1`] = `
<div
  className="button-toolbar">
  <input
    name="btn-add"
    onClick={[Function]}
    type="button"
    value="Add" />
  <input
    name="btn-clear"
    onClick={[Function]}
    type="button"
    value="Clear" />
</div>
`;
